File: adventofcode/2021/day1/part1.py

num_increased = 0
prev_num = None
with open('part1.input', 'r') as fh:
    line = fh.readline()
    prev_num = int(line)
    while line:
        # if line and line.isnumeric():
        if prev_num < int(line):
            num_increased += 1
        prev_num = int(line)
        line = fh.readline()
print(num_increased)

File: adventofcode/2021/day1/part1.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"log"
	"os"
	"strconv"
)

func main() {
	f, err := os.OpenFile("part1.input", os.O_RDONLY, os.ModePerm)
	if err != nil {
		log.Fatalf("open file error: %v", err)
		return
	}
	defer f.Close()

	rd := bufio.NewReader(f)
	for {
		line, err := rd.ReadString('\n')
		if err != nil {
			if err == io.EOF {
				break
			}

			log.Fatalf("read file line error: %v", err)
			return
		}
		depth, err := strconv.Atoi(line)
		fmt.Println(err, depth)
	}
}
